shop/src/utils/generateVariants.js

```diff
@@ -1,25 +1,42 @@
-import flatMap from 'lodash/flatMap'
+const combinationsOfElementAndArray = (element, arr) => {
+  let accumulator = []
+  if (arr.length === 0) {
+    accumulator = [...accumulator, [element, arr[0]]]
```

Comment:
Won't `arr[0]` return undefined here?

Reply:
Yes. The algorithm was written with #806 in mind, where I noted that the variant table breaks when an empty array is passed as an argument to the function 'getAllCombinations'. 

As you correctly point out, this algorithm stands in contrast to the current implementation in that it may include instances of `undefined` in the resulting 'variants' array. However, I request that we pay attention to its effect on the Admin UI:

https://user-images.githubusercontent.com/10854442/104119470-72e70380-5355-11eb-8fad-90f530966028.mov